include/envoy/http/metadata_interface.h

```diff
@@ -36,6 +37,8 @@ class MetadataMap : public UnorderedStringMap {
 
 using MetadataMapPtr = std::unique_ptr<MetadataMap>;
 
+using MetadataMapVec = std::vector<MetadataMap*>;
```

Comment:
Can we name s/MetadataMapVec/MetadaMapVector/ for consistency with other vector typedefs?

I also suspect we may want this to be a std::vector<MetadataMapPtr> so that over in #5165 as filters addMetadata, we pass ownership to the HCM and it sticks them in its metadata_map_vector_ and the ownership is clear.  What do you think?   It's a minor enough change I'm also fine waiting until you merge this into #5165 and making the change over there if that's easier.

Reply:
A quick question: you meant std::unique_ptr instead of std::vector, right? Thanks!

Reply:
Ugh, sorry, github took my std::vector\<MetadataMapPtr> and removed the content in angle braces because I didn't add a backslash.   Yeah, vector of unique pointers.

Reply:
Ah, I see. Thanks!

Reply:
Good point! Fixed.